db.go

```diff
@@ -313,6 +307,19 @@ type DB struct {
 	metrics *dbMetrics
 }
 
+// DataSource is remote source of data that can be queried.
+type DataSource interface {
+	fmt.Stringer
+	Scan(ctx context.Context, prefix string, schema *dynparquet.Schema, filter logicalplan.Expr, lastBlockTimestamp uint64, stream chan<- any) error
```

Comment:
Maybe it makes sense to pass a callback instead of a channel directly to allow the caller to choose how to process the scan results.

Reply:
Done.